raspi-config

```diff
@@ -1294,6 +1294,87 @@ do_resolution() {
   fi
 }
 
+remove_network_block() {
+  # Escape special characters to avoid screwing up regex below
+  # Escape '\' separately because it would mix up with others
+  local ssid=$(echo "$1" | sed 's;\\;\\\\;g' | sed -e 's;\.;\\\.;g'  \
+                                                   -e 's;\*;\\\*;g'  \
+                                                   -e 's;\+;\\\+;g'  \
+                                                   -e 's;\?;\\\?;g'  \
+                                                   -e 's;\^;\\\^;g'  \
+                                                   -e 's;\$;\\\$;g'  \
+                                                   -e 's;\/;\\\/;g'  \
+                                                   -e 's;\[;\\\[;g'  \
+                                                   -e 's;\];\\\];g'  \
+                                                   -e 's;{;\\{;g'    \
+                                                   -e 's;};\\};g'    \
+                                                   -e 's;(;\\(;g'    \
+                                                   -e 's;);\\);g')
+
+  # Match the following and remove it:
+  # ^\s*network=\{        ==>  network block begins
+  # (?:(?!^\s*\}.*$).)*?  ==>  no block ending in between
+  # ^\s*ssid="..."        ==>  target SSID
+  # ^\s*\}                ==>  network block ends
+  perl -i -pe 'BEGIN{undef $/;} s/^\s*network=\{(?:(?!^\s*\}.*$).)*?^\s*ssid="'$ssid'".*?^\s*\}.*?\n//smg' /etc/wpa_supplicant/wpa_supplicant.conf
+}
+
+append_nokey_network_block() {
+  local ssid=$1
+
+  cat <<EOT >> /etc/wpa_supplicant/wpa_supplicant.conf
+network={
+    ssid="$ssid"
+    key_mgmt=NONE
+}
+EOT
+}
+
+append_wpa_network_block() {
+  local ssid=$1
+  local passphrase=$2
+  local out=
+
+  if out=$(wpa_passphrase "$ssid" "$passphrase"); then
+    echo "$out" | tee -a /etc/wpa_supplicant/wpa_supplicant.conf > /dev/null
```

Comment:
` > ` will works here instead of ` | tee -a ... > /dev/null`

Reply:
I suppose you mean `>>` ... anyway, I made the change, and I also bypassed checking the presence of `wlan0` before `wpa_cli ... reconfigure`. Just do it!